File: NetCinema/src/main/java/com/cine/cinema/service/PeliculaService.java
package com.cine.cinema.service;

import com.cine.cinema.model.Pelicula;
import com.cine.cinema.repository.PeliculaRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class PeliculaService {

    private final PeliculaRepository peliculaRepository;

    // Crear película
    public Pelicula crearPelicula(Pelicula pelicula) {
        return peliculaRepository.save(pelicula);
    }

    // Obtener todas las películas
    public List<Pelicula> obtenerTodas() {
        return peliculaRepository.findAll();
    }

    // Obtener películas activas (cartelera)
    public List<Pelicula> obtenerCartelera() {
        return peliculaRepository.findByActivaTrue();
    }

    // Obtener película por ID
    public Optional<Pelicula> obtenerPorId(String id) {
        return peliculaRepository.findById(id);
    }

    // Buscar películas por título
    public List<Pelicula> buscarPorTitulo(String titulo) {
        return peliculaRepository.findByTituloContainingIgnoreCase(titulo);
    }

    // Buscar por género
    public List<Pelicula> buscarPorGenero(String genero) {
        return peliculaRepository.findByGenerosContaining(genero);
    }

    // Buscar por clasificación
    public List<Pelicula> buscarPorClasificacion(String clasificacion) {
        return peliculaRepository.findByClasificacion(clasificacion);
    }

    // Actualizar película
    public Pelicula actualizarPelicula(String id, Pelicula peliculaActualizada) {
        Pelicula pelicula = peliculaRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Película no encontrada"));

        pelicula.setTitulo(peliculaActualizada.getTitulo());
        pelicula.setDescripcion(peliculaActualizada.getDescripcion());
        pelicula.setGeneros(peliculaActualizada.getGeneros());
        pelicula.setDuracion(peliculaActualizada.getDuracion());
        pelicula.setClasificacion(peliculaActualizada.getClasificacion());
        pelicula.setDirector(peliculaActualizada.getDirector());
        pelicula.setActores(peliculaActualizada.getActores());
        pelicula.setPosterUrl(peliculaActualizada.getPosterUrl());
        pelicula.setActiva(peliculaActualizada.getActiva());

        return peliculaRepository.save(pelicula);
    }

    // Desactivar película (soft delete)
    public void desactivarPelicula(String id) {
        Pelicula pelicula = peliculaRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Película no encontrada"));

        pelicula.setActiva(false);
        peliculaRepository.save(pelicula);
    }

    // Eliminar película permanentemente
    public void eliminarPelicula(String id) {
        peliculaRepository.deleteById(id);
    }
}

File: netcinema-frontend/src/services/peliculaService.js
import api from './api';

export const peliculaService = {
    // Obtener cartelera
    getCartelera: async () => {
        const response = await api.get('/peliculas/cartelera');
        return response.data;
    },

    // Obtener todas las películas
    getAll: async () => {
        const response = await api.get('/peliculas');
        return response.data;
    },

    // Obtener película por ID
    getById: async (id) => {
        const response = await api.get(`/peliculas/${id}`);
        return response.data;
    },

    // Crear película
    create: async (pelicula) => {
        const response = await api.post('/peliculas', pelicula);
        return response.data;
    },

    // Actualizar película
    update: async (id, pelicula) => {
        const response = await api.put(`/peliculas/${id}`, pelicula);
        return response.data;
    },

    // Eliminar película
    delete: async (id) => {
        const response = await api.delete(`/peliculas/${id}`);
        return response.data;
    },

    // Buscar por título
    searchByTitle: async (titulo) => {
        const response = await api.get(`/peliculas/buscar/titulo/${titulo}`);
        return response.data;
    },
};